modules/nextflow/src/main/groovy/nextflow/datasource/SraExplorer.groovy

```diff
@@ -268,22 +319,25 @@ class SraExplorer {
         return result
     }
 
-    protected getFastqUrl(String acc) {
-        def text = readRunFastqs(acc)
+    protected getFastqUrl(String acc, List<String> queryFields) {
+        def text = readRunFastqs(acc, queryFields)
         if( !text )
             return
 
-        def lines = text.trim().readLines()
-        def files = lines[1].split(';')
-        def result = new ArrayList(files.size())
-        for( def str : files ) {
-            result.add( FileHelper.asPath("ftp://$str") )
+        if (!queryFields) {
+            def lines = text.trim().readLines()
+            def files = lines[1].split(';')
+            def result = new ArrayList(files.size())
+            for( def str : files ) {
+                result.add( FileHelper.asPath("ftp://$str") )
+            }
+
+            return result.size()==1 ? result[0] : result
```

Comment:
It's important the feature will not alter the current implementation. Therefore let's split `getFastqUrl` in two. One version does not the `queryFields` argument and does exactly the same as before. A second version implements the new stuff.

Reply:
Split here: 5015f85